PWGHF/D2H/Tasks/taskSigmac.cxx

```diff
@@ -392,9 +437,16 @@ struct HfTaskSigmac {
         }
         /// THn for candidate Σc0,++ cut variation
         if (enableTHn) {
-          /// fill it only if no MC operations are enabled, otherwise fill it in the processMC with the right origin and channel!
-          if (!doprocessMc) {
-            registry.get<THnSparse>(HIST("hnSigmaC"))->Fill(ptLc, deltaMass, decLengthLc, decLengthXYLc, cpaLc, cpaXYLc, 0, 0, ptSc, std::abs(chargeSc));
+          if (!isMc) {
+            /// fill it only if no MC operations are enabled, otherwise fill it in the processMC with the right origin and channel!
+            if constexpr (useMl) {
+              /// fill with ML information
+              /// BDT index 0: bkg score; BDT index 2: non-prompt score
+              registry.get<THnSparse>(HIST("hnSigmaC"))->Fill(ptLc, deltaMass, candidateLc.mlProbLcToPiKP()[0], candidateLc.mlProbLcToPiKP()[2], 0, 0, ptSc, std::abs(chargeSc));
```

Comment:
The vector `mlProbLcToPiKP` is filled (i.e. with non-zero size) only for `Lc` candidates that fulfil all the preselections. However if I am not wrong here you could have Lc candidates that didn't pass the preselections depending on the value of this configurable https://github.com/AliceO2Group/O2Physics/blob/f64af44dd7a98be8782a9dda39e86b5731c0483d/PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx#L49 set in the Sigmac creator. 

Wouldn't be better to check the size or the Lc flag before filling it in the THnSparse given that there is this possibility, what do you think?

Reply:
You are totally right, indeed filling with something empty (as done in the selector) != not filling.
Thanks for spotting it, I'll solve it